core/src/main/java/org/apache/iceberg/BaseManifestReader.java

```diff
@@ -218,13 +219,12 @@ public ThisT caseSensitive(boolean isCaseSensitive) {
    * @return an Iterator of DataFile. Makes defensive copies of files before returning
    */
   @Override
-  @SuppressWarnings("unchecked")
-  public CloseableIterator<T> iterator() {
+  public CloseableIterator<F> iterator() {
     if (dropStats(rowFilter, columns)) {
-      return (CloseableIterator<T>) CloseableIterable.transform(liveEntries(), e -> e.file().copyWithoutStats())
+      return CloseableIterable.transform(liveEntries(), e -> e.file().copyWithoutStats())
```

Comment:
nit: I think this can be on a single line now

Reply:
Fixed.